pandas/core/generic.py

```diff
@@ -5092,6 +5099,47 @@ def bfill(self, axis=None, inplace=False, limit=None, downcast=None):
 
         This raises a ``TypeError`` because one of the ``dict`` keys is not of
         the correct type for replacement.
+
+        Compare the behavior of
+        ``s.replace('a', None)`` and ``s.replace({'a': None})`` to understand
+        the pecularities of the ``to_replace`` parameter.
+        ``s.replace('a', None)`` is actually equivalent to
+        ``s.replace(to_replace='a', value=None, method='pad')``,
+        because when ``value=None`` and ``to_replace`` is a scalar, list or
+        tuple, ``replace`` uses the method parameter to do the replacement.
+        So this is why the 'a' values are being replaced by 30 in rows 3 and 4
+        and 'b' in row 6 in this case. However, this behaviour does not occur
+        when you use a dict as the ``to_replace`` value. In this case, it is
+        like the value(s) in the dict are equal to the value parameter.
+
+        >>> s = pd.Series([10, 20, 30, 'a', 'a', 'b', 'a'])
```

Comment:
Just to keep things concise why don't you get rid of 10 and 20 in this example? They don't serve any real purpose but make the documentation longer. Can also replace 30 with 1

Reply:
great suggestion of simplifying.